lib/data/api_service/data_api.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:dio/dio.dart';
+import 'package:marvel_app/domain/models/marvel_hero.dart';
+import 'package:marvel_app/utils/constants/api_url.dart';
+
+class DataApi {
+  final dio = Dio();
+
+  void configureDio() {
+    dio.options.baseUrl = ApiUrl.url;
+    dio.options.connectTimeout = const Duration(seconds: 5);
+    dio.options.receiveTimeout = const Duration(seconds: 5);
+  }
```

Comment:
Num app com múltiplos endpoints, o ideal seria isso aqui:
```suggestion
  void configureDio() {
    dio.options = BaseOptions(
      baseUrl: 'https://gateway.marvel.com/v1/public/',
      queryParameters: {
        'ts': '123',
        'apikey': '044479a19929992d86c53efb86add0a5',
        'hash': 'd7d2e2160357184800ff55f0b845be50',
      },
      connectTimeout: const Duration(seconds: 5),
      receiveTimeout: const Duration(seconds: 5),
    );
  }
```

Pois essas infos servirão para todos os endpoints.
Aí na chamada do endpoint em específico, seria `dio.get(characters)`